contracts/base/Escrow.sol

```diff
@@ -165,16 +157,37 @@ contract Escrow is EscrowInterface, TInitializable, BaseUpgradeable, BaseEscrowD
     }
 
     /**
-        @notice Purchase this Escrow's loan if it is eligible to liquidate and transfers ownership
-            to msg.sender.
-        @dev Events are emitted when repay() and _transferOwnership().
+        @notice Repay this Escrow's loan.
+        @dev If the Escrow's balance of the borrowed token is less than the amount to repay, transfer tokens from the sender's wallet.
+        @dev Only the owner of the Escrow can call this. If someone else wants to make a payment, they should call the loans contract directly.
      */
-    function purchaseLoanDebt() external payable {
-        require(canPurchase(), "ESCROW_INELIGIBLE_TO_PURCHASE");
+    function repay(uint256 amount) external onlyOwner {
+        IERC20 token = IERC20(loans.lendingToken());
+        uint256 balance = _balanceOf(loans.lendingToken());
+        if (amount > balance) {
+            token.transferFrom(msg.sender, address(this), amount.sub(balance));
```

Comment:
I think we should validate the return value from the transferFrom call.

Reply:
There's really no why to validate that the response will be false. The `transferFrom` function only throws if there is a problem or returns `true` so I mean I can add it but seems like it will just be redundant.